File: dart/34_pyramidArray.dart
// Challenge link: https://www.codewars.com/kata/515f51d438015969f7000013

List<List<int>> pyramidDeprecated(int n) {
  // List constructor is deprecated so new answer below
  List<List<int>> pyramidList = [];
  for (int i = 0; i < n; i++) {
    pyramidList.add(List.filled(i + 1, 1));
  }
  return pyramidList;
}

List<List<int>> pyramid(int n) {
  return List.generate(n, (int i) => List.filled(i + 1, 1));
}

File: javascript/june2021/34_pyramidArray.js
// Challenge link: https://www.codewars.com/kata/515f51d438015969f7000013

const pyramid = (n) => {
  let pyramidArr = [];
  for (let i = 0; i < n; i++) {
    pyramidArr.push(Array.from({ length: i + 1 }).map((x) => 1));
  }
  return pyramidArr;
};
